app/Controllers/User.php

```
@@ -0,0 +1,109 @@
+<?php
+
+namespace App\Controllers;
+use CodeIgniter\API\ResponseTrait;
+use App\Models\UserModel;
+
+class User extends BaseController
+{
+    use ResponseTrait;
+
+    public function index()
+    {
+        return view('pages/visitor_home');
+    }
+
+    public function login()
+    {
+        return view('pages/user_login');
+    }
+
+    public function register()
+    {
+        return view('pages/user_register');
+    }
+
+    public function home()
+    {
+        $data['memberData'] = session()->memberData;
+        return view('pages/user_home',$data);
+    }
+
+    public function doLogin()
+    {
+        $request = \Config\Services::request();
+        $data = $request->getPost();
+        // $data = $this->request->getPost();
+
+        $email = $data['email'];
+        $password = $data['password'];
+        if($email == null || $password == null){
+            return $this->respond([
```

If there are some error logic you wanna repsonse fail response, you can use `fail()` function

```php
$this->fail();
```

See https://codeigniter.com/user_guide/outgoing/api_responses.html?highlight=fail#fail

---

If you still want to use respond() to handle the fail response, you need to return the error code.

Ref https://developer.mozilla.org/en-US/docs/Web/HTTP/Status